crates/apub/src/api/resolve_object.rs

```diff
@@ -67,14 +71,154 @@ async fn convert_response(
       }
       UserOrCommunity::Community(c) => {
         removed_or_deleted = c.deleted || c.removed;
-        res.community = Some(CommunityView::read(pool, c.id, local_user.as_ref(), false).await?)
+        res.community = Some(CommunityView::read(pool, c.id, local_user.as_ref(), is_admin).await?)
       }
     },
   };
-  // if the object was deleted from database, dont return it
-  if removed_or_deleted {
+
+  // if the object was deleted from database, don't return it for regular users
+  if removed_or_deleted && !is_admin {
     Err(NotFound {}.into())
   } else {
     Ok(Json(res))
   }
 }
+
+#[cfg(test)]
+mod tests {
+  use crate::api::resolve_object::resolve_object;
+  use activitypub_federation::config::Data;
+  use actix_web::web::Query;
+  use lemmy_api_common::{context::LemmyContext, site::ResolveObject};
+  use lemmy_db_schema::{
+    newtypes::InstanceId,
+    source::{
+      community::{Community, CommunityInsertForm},
+      instance::Instance,
+      local_site::{LocalSite, LocalSiteInsertForm},
+      local_user::{LocalUser, LocalUserInsertForm},
+      person::{Person, PersonInsertForm},
+      post::{Post, PostInsertForm, PostUpdateForm},
+      site::{Site, SiteInsertForm},
+    },
+    traits::Crud,
+  };
+  use lemmy_db_views::structs::LocalUserView;
+  use lemmy_utils::{error::LemmyResult, LemmyErrorType};
+  use serial_test::serial;
+
+  async fn create_user(
+    instance_id: InstanceId,
+    name: String,
+    admin: bool,
+    context: &Data<LemmyContext>,
+  ) -> LemmyResult<LocalUserView> {
+    let person_form = PersonInsertForm::test_form(instance_id, &name);
+    let person = Person::create(&mut context.pool(), &person_form).await?;
+
+    let user_form = match admin {
+      true => LocalUserInsertForm::test_form_admin(person.id),
+      false => LocalUserInsertForm::test_form(person.id),
+    };
+    let local_user = LocalUser::create(&mut context.pool(), &user_form, vec![]).await?;
+
+    Ok(LocalUserView::read(&mut context.pool(), local_user.id).await?)
+  }
```

Comment:
There is already `create_user()` in user_settings_backup.rs so you should reuse that.

Reply:
that one works a bit differently.
would it make sense to create something like `crate::test` to store common test functions, to avoid pulling in functions from other test sections?
something like this? https://github.com/Nothing4You/lemmy/commit/2ac3d0ad5e6677b74e05819a84846dafbbc996aa

i could also add another basic wrapper in user_settings_backup.rs that'd deal with passing the instance id and admin arg.

Reply:
Yes that code looks fine. You can also make a struct like this so its not necessary to pass all the params every time:
```rust
#derive(Default) 
struct TestUser {
name: String,
instance: Instance,
...}
```

Reply:
how about this now?